templates/CMakeLists.txt.template

```diff
@@ -759,6 +820,17 @@
     set_target_properties(${lib.name} PROPERTIES COMPILE_PDB_NAME "${lib.name}"
       COMPILE_PDB_OUTPUT_DIRECTORY <%text>"${CMAKE_BINARY_DIR}</%text>"
     )
+    % if lib.name == 'gpr':
```

Comment:
to improve our cmake build, we're trying to get fewer reference to libraries by their name, not more. So in my opinion this change introduces technical debt that will need to be dealt with later (so not an ideal outcome).

Basically deciding whether a library should have some special settings assigned to it should be based on library's metadata (e.g. is the a "base" grpc public library), not by listing libraries that should have these setting by their names.

Unfortunately, the refactoring of our cmake build to extrapolate metadata like this (instead of relying of knowing the names of individual targets) is only in progress and definitely needs much more work.  My worries is that making changes like the once suggested in this PR, we'll make the subsequent refactoring even harder.

Reply:
Yes, I get it - I did look at finding a way to add that to the metadata generated from bazel, but didn't want to move before agreeing a direction. Is there a branch with the cmake build refactoring in progress? I can take at contributing if it would help.

Reply:
I have a different opinion on this. As this is considered experimental, I'd like to keep this change within CMakeLists.txt file so that others are not changed. Then it's easy to understand to follow what hacks are in place.